### DISCUSSION TITLE
How app.tsx file is picking up by nextJs first it pick app.tsx file or the page  ?

### DISCUSSION THREAD

**Discussion by manishajmera** [vercel/next.js#10816] [Help]
How app.tsx file is picking up by nextJs first it pick app.tsx file or the page  ? I want to know which file render first .

**Comment by willemliu** [comment]
The `_app.tsx` encapsulates the page. So they are picked up at the same time but you should see the page like `index.tsx` as a component rendered inside the `_app.tsx`. This all happens in the `Component` object given as `props` to your `_app.tsx`.

**Comment by manishajmera** [comment]
can i make nextJs umd bundle , it looks like it is not possible because there is not any single entry point ?

**Reply by willemliu** [reply]
Could you explain what you're trying to do?

**Reply by manishajmera** [reply]
I want to convert my nextJs app into microFrontends.

**Reply by willemliu** [reply]
Hmm OK, if you haven't already maybe you can start a new discussion describing this specific need. Maybe someone else with this knowledge will be able to find it better.

**Reply by manishajmera** [reply]
yes sure
